NewForce-Cohort-6/tabloidfullstack-cranberry-sauce
# Jg delete tag 15

# Description
[Ticket number 15](https://trello.com/c/Pf8L0dYz/15-15-delete-a-tag) added the  functionality for deleting a tag.

#Updates included. 
-client->src->components->TagList was updated to add a Delete button next to each tag that navigates users to  a confirmation page. 
-client->src->components->DeleteTag was added with an ‘Are you sure’ message  and a yes button that deletes the tag then navigates users back to the tag list and a no button that just navigates users back to the tag list. 
-client->src->Managers->TagManager was updated with `DeleteTag` function (more functions to be added later).


# Testing Instructions

1.  `git add` and `git commit` whatever you're working on on your branch
2.  `git checkout main`
3. ` git fetch --a`
4.  `git checkout jg-DeleteTag-15 ` to switch to my branch.
5. If you don't see the most recent version of the files, try` git pull origin jg-DeleteTag-15`.
6. If you haven't created the Tabloid FullStack database, please do so.
7.  Use this to fill out tables and columns: [SQL data](https://github.com/NewForce-Cohort-6/tabloidfullstack-cranberry-sauce/blob/main/SQL/01_Tabloid_Create_DB.sql)
8.  Use this to seed the database: [SQL Seed Data](https://github.com/NewForce-Cohort-6/tabloidfullstack-cranberry-sauce/blob/main/SQL/02_Tabloid_Seed_Data.sql)
9. Execute Tabloid from inside of Visual Studio by clicking on the solid green arrow. This will launch Swagger and run the back end of the application to provide the API that will be utilized to view the front end. 
10. From your terminal run npm start from the `client` directory within `tabloidfullstack-cranberry-sauce`
11.  Step 10 will bring up the front end of the application in a browser window titled React App. In the url of that browser page, please Login using an email that you know is there and works for you (example: foo@bar.com from seed data)

12. Navigate to the Tag Management page using the nav bar.  You should see a list of Tag names printed on the that include the following from the seed data: 
-C#
-Funny
-JavaScript
-Scary
13. Next to each tag name is a delete button. Select a tag that you want to delete and click the delete button next to it. 
14. You should be navigated to the page that asks if you want to delete that tag. Select Yes. 
15. You should be navigated back to the Tag List page where you should observe the absence of the tag you just deleted. 
16. Select another tag that you don’t want to delete and click the delete button next to it. 
17. From the confirmation page, click the No button. 
18. You should be navigated back to the Tag List page where you should see the tag you didn’t want to delete still there. 

Confirmation of steps 15 and 18 indicate the feature functions as intended. 



# Checklist:
